flake.nix

```diff
@@ -8,22 +8,14 @@
   inputs.nix-filter.url = "github:numtide/nix-filter";
 
   outputs = { self, nixpkgs, flake-utils, nix-filter, flake-compat }@inputs: {
+    overlay = import ./nix/overlay.nix { inherit inputs; };
     nixosModules.vast = {
-      imports = [
-        ./nix/module.nix
-        {
-          nixpkgs.config.packageOverrides = pkgs: {
-            inherit (self.packages."${pkgs.stdenv.hostPlatform.system}")
-              vast
-              ;
-          };
-        }
-      ];
+      imports = [ ./nix/module.nix ];
+      nixpkgs.overlays = [ self.overlay ];
```

Comment:
The global overlay will pollute the environment twice (default to be input.<nixos.nixpkgs_channel>. and packages). Also, I don't think the previous way made your nixops wrong. Either microvm or Nixops, I have been very well tested for it.

Reply:
I found a way to avoid one evaluation with `_module.args`. We still need one evaluation of `nixpkgs` with our overlay because we want to pin it to a specific version to ensure that the consumer gets a working binary (because it is checked in CI).